move.py

```diff
@@ -4,168 +4,137 @@
     discount_digits = price_digits
 from trytond.modules.currency.fields import Monetary
 
-__all__ = ['Move']
-
 _ZERO = Decimal(0)
 STATES = {
     'invisible': Not(Equal(Eval('state', ''), 'done')),
     }
-PARTIES = {
-    'stock.shipment.in': 'supplier',
-    'stock.shipment.in.return': 'supplier',
-    'stock.shipment.out': 'customer',
-    'stock.shipment.out.return': 'customer',
-    'stock.shipment.internal': 'company',
-    }
 
 
 class Move(metaclass=PoolMeta):
     __name__ = 'stock.move'
-    gross_unit_price = fields.Function(Monetary('Gross Price',
-        digits=price_digits, currency='currency', states=STATES),
-        'get_origin_fields')
-    amount = fields.Function(Monetary('Amount',
-        digits='currency', currency='currency'), 'get_origin_fields')
+
+    base_price = Monetary(
+        "Base Price", currency='currency', digits=price_digits,
+        states={
+            'invisible': ~Eval('unit_price_required'),
+            'readonly': Eval('state') != 'draft',
+            })
+
+    discount_rate = fields.Function(fields.Numeric(
+            "Discount Rate", digits=(16, 4),
+            states={
+                'invisible': ~Eval('unit_price_required'),
+                'readonly': Eval('state') != 'draft',
+                }),
+        'on_change_with_discount_rate', setter='set_discount_rate')
+    discount_amount = fields.Function(Monetary(
+            "Discount Amount", currency='currency', digits=price_digits,
+            states={
+                'invisible': ~Eval('unit_price_required'),
+                'readonly': Eval('state') != 'draft',
+                }),
+        'on_change_with_discount_amount', setter='set_discount_amount')
+
+    discount = fields.Function(fields.Char(
+            "Discount",
+            states={
+                'invisible': ~Eval('discount'),
+                }),
+        'on_change_with_discount')
+
     taxes = fields.Function(fields.Many2Many('account.tax', None, None,
-        'Taxes'), 'get_origin_fields')
-    unit_price_w_tax = fields.Function(Monetary('Unit Price with Tax',
-        digits='currency', currency='currency', states=STATES),
-        'get_origin_fields')
-    discount = fields.Function(Monetary('Discount',
-        digits=discount_digits, states=STATES),
-        'get_origin_fields')
-
-    def _get_tax_rule_pattern(self):
-        '''
-        Get tax rule pattern
-        '''
-        return {}
-
-    @property
-    def tax_date(self):
-        "Date to use when computing the tax"
-        pool = Pool()
-        Date = pool.get('ir.date')
-        with Transaction().set_context(company=self.company.id):
-            return Date.today()
+        'Taxes'), 'get_taxes')
+    amount = fields.Function(Monetary('Amount', digits='currency',
+        currency='currency'), 'on_change_with_amount')
 
     @classmethod
-    def get_origin_fields(cls, moves, names):
+    def __register__(cls, module_name):
+        # Rename gross_unit_price to base_price
```

Comment:
"gross_unit_price" was a function field. Why add rename column table?

Reply:
You're right, should be removed this migration.

Reply:
Done 92bc841e0b2760579314660434fdb2801fca805a